recipes/openimageio/all/conanfile.py

```diff
@@ -63,260 +64,248 @@ class OpenImageIOConan(ConanFile):
+            self.requires("libheif/1.16.2")
         if self.options.with_raw:
-            self.requires("libraw/0.20.2")
+            self.requires("libraw/0.21.2")
         if self.options.with_openjpeg:
             self.requires("openjpeg/2.5.0")
         if self.options.with_openvdb:
             self.requires("openvdb/8.0.1")
         if self.options.with_ptex:
-            self.requires("ptex/2.4.0")
+            self.requires("ptex/2.4.2")
         if self.options.with_libwebp:
-            self.requires("libwebp/1.3.1")
+            self.requires("libwebp/1.3.2")
         # TODO: R3DSDK dependency
         # TODO: Nuke dependency
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            if Version(self.version) >= "2.3.0.0" or self.options.with_openvdb:
-                check_min_cppstd(self, 14)
-            else:
-                check_min_cppstd(self, 11)
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, 14)
         if is_msvc(self) and is_msvc_static_runtime(self) and self.options.shared:
             raise ConanInvalidConfiguration(
                 "Building shared library with static runtime is not supported!"
             )
 
-    def source(self):
-        files.get(
-            self,
-            **self.conan_data["sources"][self.version],
-            strip_root=True,
-            destination=self._source_subfolder
-        )
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
-    def _patch_sources(self):
-        files.apply_conandata_patches(self)
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    @functools.lru_cache(1)
-    def _configure_cmake(self):
-        cmake = CMake(self)
+    def generate(self):
+        tc = CMakeToolchain(self)
 
         # CMake options
-        cmake.definitions["CMAKE_DEBUG_POSTFIX"] = ""  # Needed for 2.3.x.x+ versions
-        cmake.definitions["OIIO_BUILD_TOOLS"] = True
-        cmake.definitions["OIIO_BUILD_TESTS"] = False
-        cmake.definitions["BUILD_DOCS"] = False
-        cmake.definitions["INSTALL_DOCS"] = False
-        cmake.definitions["INSTALL_FONTS"] = False
-        cmake.definitions["INSTALL_CMAKE_HELPER"] = False
-        cmake.definitions["EMBEDPLUGINS"] = True
-        cmake.definitions["USE_PYTHON"] = False
-        cmake.definitions["USE_EXTERNAL_PUGIXML"] = True
+        tc.variables["CMAKE_DEBUG_POSTFIX"] = ""  # Needed for 2.3.x.x+ versions
+        tc.variables["OIIO_BUILD_TOOLS"] = True
+        tc.variables["OIIO_BUILD_TESTS"] = False
+        tc.variables["BUILD_DOCS"] = False
+        tc.variables["INSTALL_DOCS"] = False
+        tc.variables["INSTALL_FONTS"] = False
+        tc.variables["INSTALL_CMAKE_HELPER"] = False
+        tc.variables["EMBEDPLUGINS"] = True
+        tc.variables["USE_PYTHON"] = False
+        tc.variables["USE_EXTERNAL_PUGIXML"] = True
+        tc.variables["BUILD_MISSING_FMT"] = False
```

Comment:
Is robinmap a necessary or optional lib? 
The wording in https://github.com/AcademySoftwareFoundation/OpenImageIO/blob/43fea763945991d1b83f5c0bb65f12bfab09b99b/src/cmake/externalpackages.cmake#L300 makes it seem optional, and it currently is vendorized because we seem to lack that library in CCI (tracking in #22872)

Reply:
To my understanding it is needed and the code you linked allows between selecting an external one vs. having Cmake download a robinmap and build it itself. But maybe I'm missunderstanding something.

Reply:
Sorry, misunderstood that earlier. To my understanding it was by the changes and patches using tsl-robin-map as the robinmap. Not sure what the difference is if any to the MR you linked.

Reply:
So, testing is no fixed to false.